File: omr_web/src/lib/academic-term.ts
/** School year / term helpers (aligned with mobile AcademicTerm). */
export function schoolYearForDate(date = new Date()): string {
  const month = date.getMonth() + 1;
  const startYear = month >= 6 ? date.getFullYear() : date.getFullYear() - 1;
  return `${startYear}-${startYear + 1}`;
}

export const commonTermLabels = ["1st Sem", "2nd Sem", "Summer"] as const;

export function defaultTermLabel(date = new Date()): string {
  const month = date.getMonth() + 1;
  if (month >= 6 && month <= 10) return "1st Sem";
  if (month >= 11 || month <= 3) return "2nd Sem";
  return "Summer";
}

export function schoolYearOptions(past = 2, future = 1): string[] {
  const currentStart = Number.parseInt(schoolYearForDate().split("-")[0] ?? "0", 10);
  return Array.from({ length: past + 1 + future }, (_, index) => {
    const start = currentStart - past + index;
    return `${start}-${start + 1}`;
  });
}

export function formatSectionTerm(section: {
  school_year?: string | null;
  term_label?: string | null;
}): string | null {
  const parts = [section.term_label, section.school_year].filter(Boolean);
  return parts.length > 0 ? parts.join(" · ") : null;
}

File: lib/utils/academic_term.dart
/// School year and term helpers (Philippines-style: year starts around June).
class AcademicTerm {
  AcademicTerm._();

  /// e.g. "2025-2026" for SY that includes the given date.
  static String schoolYearForDate([DateTime? date]) {
    final local = (date ?? DateTime.now()).toLocal();
    final startYear = local.month >= 6 ? local.year : local.year - 1;
    return '$startYear-${startYear + 1}';
  }

  static const List<String> commonTermLabels = <String>[
    '1st Sem',
    '2nd Sem',
    'Summer',
  ];

  static String defaultTermLabel([DateTime? date]) {
    final month = (date ?? DateTime.now()).toLocal().month;
    if (month >= 6 && month <= 10) {
      return '1st Sem';
    }
    if (month >= 11 || month <= 3) {
      return '2nd Sem';
    }
    return 'Summer';
  }

  static List<String> schoolYearOptions({int past = 2, int future = 1}) {
    final currentStart = int.parse(schoolYearForDate().split('-').first);
    return List<String>.generate(
      past + 1 + future,
      (index) {
        final start = currentStart - past + index;
        return '$start-${start + 1}';
      },
    );
  }
}
